### PR TITLE
Add XRefService projects

### PR DESCRIPTION
**Pull request by vicancy**

### CODE REVIEW

**Comment by vwxyzh** [comment]
should we reference dll or create a link file instead of copy source code?

File: tools/XRefService/XRefService/Controllers/GetController.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                            orderby o.Uid
+                            select o.XRefSpecJson).ToListAsync();
+            return Json(results);
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> GetByPattern([FromQuery]string pattern, [FromQuery]int count = 100)
+        {
+            var results = await (from o in _db.XRefSpecObjects
+                               where o.Uid.Contains(pattern)
```
Comment (vwxyzh) [inline tools/XRefService/XRefService/Controllers/GetController.cs]: perf: full table scan

File: tools/XRefService/ManageXRefService/Controllers/ManageController.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            var url = request.url;
+            if (url == null)
+            {
+                return BadRequest();
+            }
+
+            var tasks = new List<Task>();
+            using (var client = new WebClient())
+            {
+                var result = await client.DownloadStringTaskAsync(new System.Uri(url));
```
Comment (vwxyzh) [inline tools/XRefService/ManageXRefService/Controllers/ManageController.cs]: use stream

File: tools/XRefService/ManageXRefService/Controllers/ManageController.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        }
+
+        [HttpPost("uploads")]
+        public async Task<IActionResult> UploadXRefMap([FromBody]UploadXRefMapRequest request)
+        {
+            var url = request.url;
+            if (url == null)
+            {
+                return BadRequest();
+            }
```
Comment (vwxyzh) [inline tools/XRefService/ManageXRefService/Controllers/ManageController.cs]: return accept?

File: tools/XRefService/ManageXRefService/Controllers/ManageController.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+            var tasks = new List<Task>();
```
Comment (vwxyzh) [inline tools/XRefService/ManageXRefService/Controllers/ManageController.cs]: useless?

File: tools/XRefService/XRefService/Controllers/GetController.cs (hunk 1 above)
Comment (vicancy) [reply tools/XRefService/XRefService/Controllers/GetController.cs]: Add a todo for full text search feature

**Pull request merged**